tests/integration/api/users/GroupSearchTest.php

```diff
@@ -0,0 +1,331 @@
+        $this->assertObjectNotHasAttribute('included', $responseBodyContents, json_encode($responseBodyContents));
+    }
+
+    /**
+     * @test
+     */
+    public function non_admin_gets_correct_results_group_id()
+    {
+        $this->prepareDatabase([
+            'group_permission' => [
+                ['permission' => 'viewUserList', 'group_id' => 2],
+            ],
+        ]);
+
+        $response = $this->send(
+            $this->request('GET', '/api/users')
+                ->withQueryParams(['filter' => ['q' => 'group:1']])
+        );
+        $responseBodyContents = json_decode($response->getBody()->getContents());
+
+        $this->assertCount(1, $responseBodyContents->data, json_encode($responseBodyContents));
+        $this->assertCount(1, $responseBodyContents->included, json_encode($responseBodyContents));
+        $this->assertEquals(1, $responseBodyContents->included[0]->id);
+
+        $response = $this->send(
+            $this->request('GET', '/api/users')
+                ->withQueryParams(['filter' => ['q' => 'group:4']])
+        );
+        $responseBodyContents = json_decode($response->getBody()->getContents());
+
+        $this->assertCount(0, $responseBodyContents->data, json_encode($responseBodyContents));
+        $this->assertObjectNotHasAttribute('included', $responseBodyContents, json_encode($responseBodyContents));
+    }
+
+    /**
+     * @test
+     */
+    public function non_admin_cannot_see_hidden_groups()
+    {
+        $this->prepareDatabase([
+            'users' => [
+                [
+                    'id' => 3,
+                    'username' => 'normal2',
+                    'password' => '$2y$10$LO59tiT7uggl6Oe23o/O6.utnF6ipngYjvMvaxo1TciKqBttDNKim', // BCrypt hash for "too-obscure"
+                    'email' => 'normal2@machine.local',
+                    'is_email_confirmed' => 1,
+                ],
+            ],
+            'groups' => [
+                [
+                    'id' => 99,
+                    'name_singular' => 'hidden user',
+                    'name_plural' => 'hidden users',
+                    'is_hidden' => true
+                ],
+            ],
+            'group_user' => [
+                [
+                    'user_id' => 3,
+                    'group_id' => 99
+                ]
+            ],
+        ]);
+        $response = $this->send(
+            $this->request('GET', '/api/users', [
+                'authenticatedAs' => 2
+            ])->withQueryParams(['filter' => ['q' => 'group:99']])
+        );
+        $responseBodyContents = json_decode($response->getBody()->getContents());
+
+        $this->assertCount(0, $responseBodyContents->data, json_encode($responseBodyContents));
+        $this->assertObjectNotHasAttribute('included', $responseBodyContents, json_encode($responseBodyContents));
+    }
+
+    /**
+     * @test
+     */
+    public function admin_gets_correct_results_group_name_singular()
```

Comment:
These cases and the "non admin" cases could probably be condensed into "authorized_user_gets_correct_results". The hidden group test case is an important distinction and should be kept.